_posts/fr/2019-03-13-iam-aws.md

```diff
@@ -0,0 +1,148 @@
+
+## Authentification
+
+Il existe 3 types d'authentification sur IAM.
+
+La première, Username/Password, est la plus simple elle permet de se connecter via un login/password. IAM conseille de mettre en place une politique de mot de passe personnalisé. Vous pouvez alors choisir plusieurs options :
+- Nombre de caractères
+- Nombre de caractères numériques
+- Changement de mot de passe tous les X jours
+- etc ...
+
+La seconde est access key/secret key. Il s'agit d'une paire clé/secret qui permet de se connecter au SDK AWS. Cela vous permet d'utiiser AWS cli ainsi que l'api REST disponible par AWS.
+
+La dernière est un mélange de access key et de session key. Elle est utilisée dans le cas des rôles et permet à l'utilisateur de se servir des services AWS via l'api REST.
+
+## Autorisation
+
+Une fois l'authentification effectuée sur IAM, il faut donner des accès à vos utilisateurs. C'est dans ce cadre que l'autorisation est appliquée. Il s'agit alors de donner des accès spécifiques à chaque utilisateur. Pour ce faire nous utlisons les "policies" qui permettent de donner des droits très précis sur les services.
+
+### Policies
+
+Il existe déjà beaucoup de "policies" crées par AWS. Vous pouvez choisir très rapidement des "policies" du type :
+- Lecture sur Bucket
+- Écriture Lambda
+
+Mais il est possible de créer sa propre "policies". Pour cela il faut créer un objet JSON. Cet objet doit contenir les infos suivantes :
+
+- `Effect` qui contient la valeur `Allow` ou `Deny`
+- `Resource` qui contient l'ARN de la ressource utilisée sur la "policy"
+- `Action` qui est une liste d'actions possibles representée par `Service:Action`
+- `Condition` qui permet de mettre une condition à notre "policy", exemple "l'IP est..."
+
+Cela pourrait donner un objet tel que :
+
+```json
+{
+"Version": "2012-10-17",
+"Statement": [
+{
+"Sid": "Stmt2649856320145",
+"Effect": "Allow",
+"Action": [
+"s3:GetObject",
+"s3:ListBucket"
+],
+"Condition": {
+"IpAddress": {
+"aws:SourceIp": "192.168.0.1"
+}
+},
+"Resource": [
+"arn:aws:s3::my_public_bucket/*"
+]
+}
+]
+}
+```
+
+Une fois votre "policy" créé, vous pouvez la mettre soit sur un utilisateur IAM, soit sur un groupe.
+
+## Autres features
+
+Bien sûr, IAM propose d'autres fonctionnalités très importantes.
+La première est la gestion du MFA (Multi-factor authentification), qui permet d'utliser une autre source d'authentification pour un utilisateur IAM. Vous pouvez très facilement ajouter la double authentification via Authenticator Google.
+
+La seconde fonctionnalité est la rotation des clés d'accès. Le veillissement d'une clé d'accès augmente l'insécurité, elle peut ne plus être utilisée ou alors être utilisée sur des machines anciennes. Via AWS il est possible de générer une nouvelle clé pour un utilisateur IAM. Cela permet de désactiver l'ancienne et d'en créer une nouvelle avec les mêmes droits.
+
```

Comment:
Il y a aussi l'integration avec les plus gros sso (saml) comme FB, google ect...
https://aws.amazon.com/fr/blogs/security/how-to-set-up-federated-single-sign-on-to-aws-using-google-apps/